### PR TITLE
annotate builds on clone

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/4709
also fixes https://github.com/openshift/origin/issues/3368

### CODE REVIEW

**Comment by bparees** [comment]
@jwforres fyi, @csrwng ptal.

**Comment by bparees** [comment]
[test]

File: pkg/build/generator/generator.go [hunk 1]
```diff
@@ -508,9 +526,9 @@ func (g *BuildGenerator) resolveImageSecret(ctx kapi.Context, secrets []kapi.Sec
 }
 
 // getNextBuildName returns name of the next build and increments BuildConfig's LastVersion.
-func getNextBuildName(bc *buildapi.BuildConfig) string {
-	bc.Status.LastVersion++
-	return fmt.Sprintf("%s-%d", bc.Name, bc.Status.LastVersion)
+func getNextBuildName(buildConfig *buildapi.BuildConfig) string {
+	buildConfig.Status.LastVersion++
+	return fmt.Sprintf("%s-%d", buildConfig.Name, buildConfig.Status.LastVersion)
```
Comment (csrwng) [inline pkg/build/generator/generator.go]: Just curious, what was wrong with having 'bc' here?

**Comment by mfojtik** [comment]
LGTM

File: pkg/build/generator/generator.go (hunk 1 above)
Comment (mfojtik) [inline pkg/build/generator/generator.go]: I was just going to ask the same question :-)

**Comment by csrwng** [comment]
LGTM

**Comment by mfojtik** [comment]
[merge]

File: pkg/build/generator/generator.go (hunk 1 above)
Comment (bparees) [inline pkg/build/generator/generator.go]: absolutely nothing.  i think i wrote some new code using buildConfiguration and then that lead to refactoring and i just started changing it.  fundamentally i think it reads better, though, especially for someone not used to working in this code.

**Pull request merged**